src/oop/programs_with_multiple_classes/medium/q5/MainTest.java

```diff
@@ -0,0 +1,143 @@
+        assertTrue(monitor.hasModifier(classArguments, "public"),
+                "Your " + monitorClass + " constructor does not have the correct visibility modifier.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("monitorInputProvider")
+    public void monitorConstructorInitializesValuesCorrectly(int value1, boolean value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, int.class},
+                {value2, boolean.class}
+        };
+        Object chairInstance = monitor.createInstance(arguments);
+        _assertEquals(value1, monitor.getFieldValue(chairInstance, varCount),
+                "Your " + monitorClass + " constructor does not correctly initialize the " + varCount + " attribute.");
+        _assertEquals(value2, monitor.getFieldValue(chairInstance, varOn),
+                "Your " + monitorClass + " constructor does not correctly initialize the " + varOn + " attribute.");
+    }
+
+    @Test
+    public void deskClassHasRequiredConstructor() {
+        Class<?>[] classArguments = {String.class};
+        assertTrue(desk.hasConstructor(classArguments),
+                "Your " + deskClass + " constructor does not have the correct parameters.");
+        assertTrue(desk.hasModifier(classArguments, "public"),
+                "Your " + deskClass + " constructor does not have the correct visibility modifier.");
+    }
+
+    private static Stream<Arguments> deskInputProvider() {
+        return Stream.of(
+                Arguments.of("Mialani"),
+                Arguments.of("Oak"),
+                Arguments.of("Steel")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("deskInputProvider")
+    public void deskConstructorInitializesValuesCorrectly(String material) throws Throwable {
+        Object[][] arguments = {
+                {material, String.class},
+        };
+        Object chairInstance = desk.createInstance(arguments);
+        _assertEquals(material, desk.getFieldValue(chairInstance, varMaterial),
+                "Your " + deskClass + " constructor does not correctly initialize the " + varMaterial + " attribute.");
+        _assertArrayEquals(new Monitor[2], desk.getFieldValue(chairInstance, varMonitors),
+                "Your " + deskClass + " constructor does not correctly initialize the " + varMonitors + " attribute.");
+    }
+
+    private static Stream<Arguments> monitorInputProvider() {
+        return Stream.of(
+                Arguments.of(11, true),
+                Arguments.of(20, false),
+                Arguments.of(43, true)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("monitorInputProvider")
+    public void correctFallenMethod(int count, boolean on) throws Throwable {
+        Object[][] arguments = {
+                {count, int.class},
+                {on, boolean.class}
+        };
+        Object monitorInstance = monitor.createInstance(arguments);
+        assertTrue(monitor.hasMethod(fallenMethod, null, Void.TYPE),
```

Comment:
Please separate these out into different asserts. You should check the method header, and return type separately